File: src/orchestrator/api/schemas/routines.py
"""Routine API schemas."""

from orchestrator.api.schemas.base import ApiModel


class StepSummarySchema(ApiModel):
    id: str
    title: str
    task_count: int


class RoutineSummary(ApiModel):
    id: str
    name: str
    description: str | None = None
    source: str
    step_count: int
    input_count: int
    is_archived: bool = False


class RoutineDetail(ApiModel):
    id: str
    name: str
    description: str | None = None
    source: str
    inputs: list[dict[str, object]]
    steps: list[StepSummarySchema]
    is_archived: bool = False


class ArchiveRoutineResponse(ApiModel):
    id: str
    source: str
    is_archived: bool


class RoutineListResponse(ApiModel):
    routines: list[RoutineSummary]


class ValidateRoutineRequest(ApiModel):
    yaml_content: str


class ValidateRoutineResponse(ApiModel):
    valid: bool
    errors: list[str] = []
    builder_feedback: list[str] = []

File: ui/src/types/routines.ts
export interface StepSummarySchema {
  id: string;
  title: string;
  task_count: number;
}

export interface RoutineSummary {
  id: string;
  name: string;
  description: string | null;
  source: string;
  step_count: number;
  input_count: number;
  is_archived: boolean;
}

export interface RoutineDetail {
  id: string;
  name: string;
  description: string | null;
  source: string;
  inputs: Record<string, unknown>[];
  steps: StepSummarySchema[];
  is_archived: boolean;
}

export interface ArchiveRoutineResponse {
  id: string;
  source: string;
  is_archived: boolean;
}

export interface RoutineListResponse {
  routines: RoutineSummary[];
}
